Neos.ContentRepository.Core/Classes/Feature/NodeReferencing/Dto/NodeReferencesToWrite.php

```diff
@@ -47,32 +66,34 @@ public static function fromReferences(array $references): self
     }
 
     /**
-     * @param array<int,array<string,mixed>> $values
+     * @param array<string, array<array{"referenceName": string, "target": string, "properties"?: array<string, mixed>}>> $namesAndReferences
```

Comment:
im not sure if we really like to start annotating the fromArray shape as its a lost battle in terms of time to invest :D But thanks either way ^^